internal/github/checkPendingCI.go

```diff
@@ -0,0 +1,66 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v60/github" // Ensure your go-github library version matches
+
+	"github.com/chia-network/github-bot/internal/config"
+)
+
+// CheckForPendingCI will return a list of PR URLs that are ready for CI to run but haven't started yet.
+func CheckForPendingCI(githubClient *github.Client, internalTeam string, cfg config.LabelConfig) ([]string, error) {
+	teamMembers, _ := GetTeamMemberList(githubClient, internalTeam)
+	var pendingPRs []string
+	for _, fullRepo := range cfg.LabelCheckRepos {
+		log.Println("Checking repository:", fullRepo.Name)
+		parts := strings.Split(fullRepo.Name, "/")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid repository name - must contain owner and repository: %s", fullRepo.Name)
+		}
+		owner := parts[0]
+		repo := parts[1]
+
+		// Fetch community PRs using the FindCommunityPRs function
+		communityPRs, err := FindCommunityPRs(owner, repo, teamMembers, githubClient)
+		if err != nil {
+			return nil, err
+		}
+
+		// Now check if they've been updated in the last 2 hours and are awaiting CI actions
+		cutoffTime := time.Now().Add(-2 * time.Hour) // 2 hours ago
+
+		for _, pr := range communityPRs {
+			if pr.CreatedAt.After(cutoffTime) {
```

Comment:
This would get PRs as they're created but I think ignores community PRs that have had new commits published to them that are awaiting maintainer approval for CI.

Reply:
I added a check to see if there was a dismissal for a prior approval. Also, if a commit is made it will change the cutofftime.